enterprise_catalog/apps/api/tasks.py

```diff
@@ -737,9 +761,13 @@ def _get_algolia_products_for_batch(
     catalog_uuids_by_key = defaultdict(set)
     customer_uuids_by_key = defaultdict(set)
     catalog_queries_by_key = defaultdict(set)
+    academy_uuids_by_key = defaultdict(set)
+    academy_tags_by_key = defaultdict(set)
 
     catalog_query_uuid_by_catalog_uuid = defaultdict(set)
     customer_uuid_by_catalog_uuid = defaultdict(set)
+    academy_uuids_by_catalog_uuid = defaultdict(set)
+    academy_tags_by_catalog_uuid = defaultdict(set)
 
     # Create a shared convenience queryset to prefetch catalogs for all metadata lookups below.
     all_catalog_queries = CatalogQuery.objects.prefetch_related('enterprise_catalogs')
```

Comment:
I think this should become:
```suggestion
    all_catalog_queries = CatalogQuery.objects.prefetch_related('enterprise_catalogs', 'enterprise_catalogs__academies')
```

Reply:
hopefully this prevents SQL fetches below inside the first pass when you retrieve the academy of a catalog.

Reply:
Done